packages/seo-integration/src/hooks/useReplacementVariables.js

```diff
@@ -0,0 +1,39 @@
+import { useSelect } from "@wordpress/data";
+import { useRef } from "@wordpress/element";
+import { SEO_STORE_NAME } from "@yoast/seo-store";
+import { filter, get, map } from "lodash";
+import { useSeoContext } from "../seo-context";
+
+/**
+ * Formats the replacement variables.
+ *
+ * Caches the given replacement variables so that the memory address only changes when any value changes.
+ *
+ * @returns {{replacementVariables: {name: string, label: string, value: *}[], recommendedReplacementVariables: string[]}} The replacement variables
+ *     and recommended replacement variables.
+ */
+export const useReplacementVariables = () => {
+	const analysisType = useSelect( select => select( SEO_STORE_NAME ).selectAnalysisType() );
+	const { analysisTypeReplacementVariables } = useSeoContext();
+	const replacementVariables = get( analysisTypeReplacementVariables, `${ analysisType }.variables`, [] );
+
+	const cache = useRef();
+
+	const values = map( replacementVariables, variable => variable.getReplacement() ).join( "" );
+	const cachedValues = map( cache.current?.replacementVariables, "value" ).join( "" );
```

Comment:
Not part of this PR, but using `map` with a string parameter I personally find very confusing (or, to frame it more positively, TIL this was actually possible 😄). Later versions of Lodash have `pluck` which seems a more apt function.

Reply:
I agree. I always opt to use JavaScript's native `map` method as defined on arrays:
```js
cache.current?.replacementVariables.map( variable => variable.name ).join( "" );
```
but perhaps that is just personal preference 😄 .